pycytominer/tests/test_cyto_utils/test_cells.py

```diff
@@ -429,10 +437,102 @@ def test_merge_single_cells_subsample():
         )
 
 
+def test_merge_single_cells_annotate():
+    """
+    Tests SingleCells.merge_single_cells using optional annotate functionality
+    """
+
+    expected_sc_merged_df = annotate(
+        profiles=AP.merge_single_cells(),
+        platemap=PLATEMAP_DF,
+        join_on=["Metadata_well_position", "Metadata_Well"],
+    )
+    sc_merged_df = AP.merge_single_cells(
+        platemap=PLATEMAP_DF, join_on=["Metadata_well_position", "Metadata_Well"]
+    )
+
+    pd.testing.assert_frame_equal(sc_merged_df, expected_sc_merged_df)
```

Comment:
I'm a bit confused how this is working. I don't see any annotate arguments in line 450?

Reply:
With these changes `merge_single_cells` will attempt to pass kwargs to `annotate` when the platemap argument receives a dataframe, using the `merge_single_cells` dataframe result as the `annotate.profiles` argument dataframe (within the `SingleCells` method merging occurs first, then annotation). This test seeks to demonstrate an isolated/independent `merge_single_cells` + `annotate` result along with an "embedded" version of the same thing.

Reply:
Gotcha!